### PR TITLE
Added broadcast invalidate status counts to processes

### PR DESCRIPTION
**Pull request by Georgi2704**
From issue:

https://github.com/workfloworchestrator/orchestrator-ui-library/issues/1411

### CODE REVIEW

File: orchestrator/workflow.py [hunk 1]
```diff
@@ -1393,6 +1404,7 @@ def _exec_steps(steps: StepList, starting_process: Process, dblogstep: StepLogFu
         # as bare exceptions are not JSON serializable
         result_to_log = step_result_process.on_failed(error_state_to_dict).on_waiting(error_state_to_dict)
         result_to_log.on_success(mutationlogger).on_failed(errorlogger).on_waiting(errorlogger)
+        result_to_log.on_failed(broadcast_failed_step)
```
Comment (Mark90) [inline orchestrator/workflow.py:1407]: Nice. We also need to run at after successful completion of a process though? For when a retried process transitions from failed to successful.

Which makes me think that we might as well call `broadcast_failed_step()` one level up in runwf() after it returns from `_exec_steps(steps, next_state, _logstep)`.

Comment (Mark90) [reply orchestrator/workflow.py:1407]: I like @tjeerddie suggestion better to do it on the resume endpoint + here on step failure. Then we don't need what I suggested

**Review by tjeerddie — APPROVED** [review]

File: orchestrator/websocket/__init__.py [hunk 2]
```diff
@@ -105,6 +105,16 @@ async def invalidate_subscription_cache(subscription_id: UUID | UUIDstr, invalid
     await broadcast_invalidate_cache({"type": "subscriptions", "id": str(subscription_id)})
 
 
+def broadcast_invalidate_status_counts() -> None:
+    """Broadcast data of the current process to connected websocket clients."""
```
Comment (Mark90) [inline orchestrator/websocket/__init__.py:109]: Can you update this docstring?

File: orchestrator/api/api_v1/endpoints/processes.py [hunk 3]
```diff
@@ -115,6 +120,7 @@ def delete(process_id: UUID) -> None:
     if not process:
         raise_status(HTTPStatus.NOT_FOUND)
 
+    broadcast_invalidate_status_counts()
```
Comment (Mark90) [inline orchestrator/api/api_v1/endpoints/processes.py:123]: I wonder why the old broadcast statement was done above the database commit. Do you have any idea perhaps @tjeerddie 

I think we should move them to below the commit, to ensure that the broadcasting action can never delay the (more important) DB action, and eliminate the (very small) possibility that the UI has done something on the broadcasted event _before_ the DB action completed.

Comment (Mark90) [reply orchestrator/api/api_v1/endpoints/processes.py:123]: (This would apply to all other broadcast statements as well)

Comment (tjeerddie) [reply orchestrator/api/api_v1/endpoints/processes.py:123]: yea, haven't reviewed previous MR's strict enough, I agree that they should happen after DB actions

File: docs/reference-docs/websockets.md [hunk 4]
```diff
@@ -46,3 +46,62 @@ Backend processes are executed in a threadpool and therefore access the same WSM
 To solve this there is a dedicated `ProcessDataBroadcastThread` (attached to and managed by the `OrchestratorCore` app) to perform the actual `broadcast_data()` call.
 
 The API endpoints which start/resume/abort a process, call `api_broadcast_process_data(request)` to acquire a function that can be used to submit process updates into a `threading.Queue` on which `ProcessDataBroadcastThread` listens.
+
```
Comment (Georgi2704) [inline docs/reference-docs/websockets.md:49]: Nice addtion !

**Pull request merged**